scripts/tests/test_upgrade.sh

```diff
@@ -19,15 +20,42 @@ then
     NUM_VALS=2
 fi
 
+echo "INFO: Building binary with upgraded version"
 cd $HOME
 export REPO=$(basename $GH_URL .git)
-rm -rf $REPO
-git clone $GH_URL && cd $REPO
-git fetch && git checkout $UPGRADE_VERSION
+if [ ! -d $REPO ]
+then
+    git clone $GH_URL
+fi
+cd $REPO
+git fetch --all && git checkout $UPGRADE_VERSION
 make build
 for (( a=1; a<=$NUM_VALS; a++ ))
 do
     export DAEMON_HOME_$a=$DAEMON_HOME-$a
     mkdir -p "$DAEMON_HOME-$a"/cosmovisor/upgrades/$UPGRADE_NAME/bin
     cp ~/$REPO/build/$DAEMON "$DAEMON_HOME-$a"/cosmovisor/upgrades/$UPGRADE_NAME/bin/
 done
+
+CURRENT_BLOCK_HEIGHT=$($DAEMON status --node $RPC | jq '.SyncInfo.latest_block_height|tonumber')
+
+echo "INFO: Submitting software upgrade proposal for upgrade: $UPGRADE_NAME"
+$DAEMON tx gov submit-proposal software-upgrade $UPGRADE_NAME --title $UPGRADE_NAME \
+    --description upgrade --upgrade-height $((CURRENT_BLOCK_HEIGHT + 60)) --deposit 10000000$DENOM \
+    --from validator1 --yes --keyring-backend test --home $DAEMON_HOME-1 --node $RPC --chain-id $CHAINID
+
+sleep 3s
+
+echo "INFO: Voting on created proposal"
+$DAEMON tx gov vote 1 yes --from validator1 --yes --keyring-backend test \
+    --home $DAEMON_HOME-1 --node $RPC --chain-id $CHAINID
+
+
+echo "INFO: Waiting for proposal to pass and upgrade"
+sleep 60s
+
+# moving back to current file folder
+cd $CURPATH
+
```

Comment:
we should query the node version before executing the tests. It will ensure that we are executing txs/queries against the correct upgraded binary and not the old one. To query node version,

```
curl -s "http://127.0.0.1:26657/abci_query?path=%22/app/version%22" | jq -r '.result.response.value' | base64 -d && echo
```